src/core/events/dispatcher.ts

```diff
@@ -62,7 +62,11 @@ export default class Dispatcher {
     var callbacks = this.callbacks.get(eventName);
     if (callbacks && callbacks.length > 0) {
       for (i = 0; i < callbacks.length; i++) {
-        callbacks[i].fn.call(callbacks[i].context || global, data);
+        if(metadata) {
+          callbacks[i].fn.call(callbacks[i].context || global, data, metadata);
+        } else {
+          callbacks[i].fn.call(callbacks[i].context || global, data);
```

Comment:
As far as I understand, this is the same as the expression on line 66, so I suggest removing the `if` condition and the `else` block.

Reply:
I changed this _slightly_ https://github.com/pusher/pusher-js/pull/323/commits/9ef0f7b90018a6adb36f04373748ee6fe326f75a.

I think using an `args` array is nicer and doesn't break any tests!